src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1139,9 +1141,9 @@ bool GenerateAnchors()
                                     Ldloc(_runtextposLocal);
                                     Ldloc(_runtextendLocal);
                                     Beq(l2);
-                                    Ldthisfld(s_runtextField);
+                                    Ldloc(_runtextSpanLocal);
```

Comment:
Does this work?  I would have expected this to need to be `Ldloca`?

Reply:
mm I see your point, I ran all tests and they were passing so I assumed it was working but I'm not sure if the tests are touching this line in particular. Also, I only just learned about the difference between `Ldloc` and `Ldloca` 😆. Will change and do some manual testing to ensure things are working as expected.

Reply:
hopefully we can find a way to add a test too. I thought we had near perfect code coverage of this code, but that of course doesn't mean that the emitted code is getting tested itself..

Reply:
I think all the test failures suggest we do actually have coverage ;-)
e.g.
```
      System.InvalidProgramException : The JIT compiler encountered invalid IL code or an internal limitation.
      Stack Trace:
           at Regex7_FindFirstChar(RegexRunner)
           at System.Text.RegularExpressions.CompiledRegexRunner.FindFirstChar()
           at System.Text.RegularExpressions.RegexRunner.ScanInternal[TState](Regex regex, String text, Int32 textstart, TState& state, MatchCallback`1 callback, Boolean reuseMatchObject, TimeSpan timeout)
           at System.Text.RegularExpressions.Regex.Run[TState](String input, Int32 startat, TState& state, MatchCallback`1 callback, Boolean reuseMatchObject)
           at System.Text.RegularExpressions.Regex.Split(Regex regex, String input, Int32 count, Int32 startat)
           at System.Text.RegularExpressions.Regex.Split(String input)
           at System.Text.RegularExpressions.Tests.RegexSplitTests.Split(RegexEngine engine, String pattern, String input, RegexOptions options, Int32 count, Int32 start, String[] expected)
```

Reply:
Oops looks like I forgot that running dotnet test on the unit test project doesn't rebuild the src and I may have forgotten to do so manually. Will fix this and push the change.

Reply:
Ok, pushed the changes. This time I did double check that all tests passed locally 😆